layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -319,6 +319,14 @@ initialized with the current filename."
                (new-name (read-file-name "New name: " (if arg dir filename))))
           (cond ((get-buffer new-name)
                  (error "A buffer named '%s' already exists!" new-name))
+                ((string-equal new-name filename)
+                 (message
+                  (propertize
+                   (format "File renaming failed! Same new and old name: '%s'"
+                           name)
+                   'face 'warning))
+                 (sit-for 2)
+                 (spacemacs/rename-current-buffer-file))
```

Comment:
Do we really need to call this function again? And if yes, maybe we should pass `arg` to it?

Reply:
The reason for calling it again was to stay at the rename prompt, and let us type a new name directly, without having to open the rename prompt manually again.

If by `arg` you mean the current optional argument that's named `arg`, then that only seems to determine if just the `dir` (path without the file and extension name) or the `filename` (full path) should be shown after "New name: " at the rename prompt.
```elisp
               (new-name (read-file-name "New name: " (if arg dir filename))))
```

Or if you meant that we could add a second optional argument, then that seems like a possible solution.

It makes it possible call the rename function with `nil` for `arg` and for example:
```text
"Rename failed! Same new and old name; "
```
(maybe there's a shorter name that says the same thing?) as the second argument.

That results in the rename prompt changing from:
```text
New name: c:/Users/username/temp/file.txt"
```
to
```text
Rename failed! Same new and old name; New name: c:/Users/username/temp/file.txt"
```
when `RET` is pressed.

It would have been great to be able to show the rename failed text with one of the faces: warning (orange) or error (red), but it seems like the `read-file-name` function doesn't show the color of a propertized string.

One issue is if the Emacs frame or helm buffer has a narrow width, then the start of the rename prompts text scrolls off to the left.

---

So there's at least three outcomes when pressing `RET` after trying to rename to the current name:
- Close the rename helm prompt, show the rename failed message in the minibuffer for 2 seconds then open the rename prompt again.
- Close the rename prompt, show the rename failed message in the minibuffer.
- (With a second optional argument), keep the rename prompt open but change its text.